packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -296,12 +295,12 @@ const reverse = (keywordRecord = {}) => {
  * @param {ERef<Invitation<Result, Args>>} invitation
  *   Invitation to contractB
  *
- * @param {KeywordKeywordRecord=} keywordMapping
+ * @param {KeywordKeywordRecord} keywordMapping
```

Comment:
I'm surprised that TS lets you get away with this, but I consider it incorrect. It does not describe the type that the caller needs to provide. Grepping reveals at least four callers that provide `undefined` in this argument position, which works because the parameter is indeed, at the JS level, declared optional using the JS optional parameter syntax. 

I'll make incremental suggestions that preserve what I take to be your intent without making this parameter optional at either the TS or JS levels.
```suggestion
 * @param {KeywordKeywordRecord | undefined} keywordMapping
```
accurately tells callers what they can provide in this position. With the next change, it also becomes an accurate description of what value is received at the moment the function is entered.

Reply:
Although I do not recommend changing it, I'll note that this many differentiated parameters is too many to be passed positionally. We should have passed a record.

Reply:
I thought the meaning of 'optional' in javascript was that you didn't have to provide a value in the call. It's pretty clear that if the 5th positional parameter is required, then you have to provide a 4th parameter (even if it's `undefined`) in order to have a 5th parameter. 

Does this interpretation mean that it's only legal to provide `undefined` as an argument if the parameter is declared optional?

Reply:
> Does this interpretation mean that it's only legal to provide `undefined` as an argument if the parameter is declared optional?

No. But the best way to declare this for a non-optional parameter is the way you already have, with a `| undefined`. I just want to reserve syntax that suggests "optional parameter" only for truly optional parameters.